src/app/filter/filter.component.ts

```diff
@@ -0,0 +1,35 @@
+import { Component } from '@angular/core';
+import { FormControl, FormGroup } from '@angular/forms';
+import { Filters, FiltersFormGroup } from './filter.model';
+import { FilterService } from './filter.service';
+
+@Component({
+  selector: 'app-filter',
+  templateUrl: './filter.component.html',
+  styleUrls: ['./filter.component.scss']
+})
+export class FilterComponent {
+  filterForm: FormGroup<FiltersFormGroup>;
+
+  constructor(private readonly filterService: FilterService) {
+    const { currentPage, entriesPerPage, nameFilter }: Filters = this.filterService.values$.value;
+
+    this.filterForm = new FormGroup<FiltersFormGroup>({
+      currentPage: new FormControl(currentPage),
+      entriesPerPage: new FormControl(entriesPerPage),
+      nameFilter: new FormControl(nameFilter)
+    });
+  }
+
+  updateFilters(): void {
+    const { currentPage, entriesPerPage, nameFilter } = this.filterForm.value;
+
+    if (currentPage != undefined && entriesPerPage != undefined) {
+      this.filterService.setState({
+        currentPage,
+        entriesPerPage,
+        nameFilter: nameFilter ?? undefined
```

Comment:
```suggestion
        nameFilter,
```

why not this directly?

Reply:
It would be a type mismatch as the value of the FormControl could be null but the type for the filter state doesn't accept it

Reply:
you are like `strictNullCheck`